contracts/validator-manager/PoSValidatorManager.sol

```diff
@@ -265,12 +275,100 @@ abstract contract PoSValidatorManager is
+     * or the beginning of the validation period, whichever is later. Reward eligibility is determined by the
+     * submitted uptime proof.
+     *
+     *
+     * @dev See {IPoSValidatorManager-claimValidationRewards}.
+     */
+    function claimValidationRewards(
+        bytes32 validationID,
+        uint32 messageIndex
+    ) external nonReentrant {
+        PoSValidatorManagerStorage storage $ = _getPoSValidatorManagerStorage();
+
+        Validator memory validator = getValidator(validationID);
+        if (validator.status != ValidatorStatus.Active) {
+            revert InvalidValidatorStatus(validator.status);
+        }
+
+        // Non-PoS validators are required to boostrap the network, but are not eligible for rewards.
+        if (!_isPoSValidator(validationID)) {
+            revert ValidatorNotPoS(validationID);
+        }
+
+        // Rewards can only be claimed by the validator owner.
+        if ($._posValidatorInfo[validationID].owner != _msgSender()) {
+            revert UnauthorizedOwner(_msgSender());
+        }
+
+        // Check that minimum stake duration has passed.
+        uint64 claimTime = uint64(block.timestamp);
+        if (claimTime < validator.startedAt + $._posValidatorInfo[validationID].minStakeDuration) {
+            revert MinStakeDurationNotPassed(claimTime);
+        }
+
+        // The claim's uptime is the difference between the total uptime and the minimum possible uptime from the last claim.
+        // We use the minimum uptime to get a lower bound on the required uptime for this claim
+        uint64 totalUptime = _updateUptime(validationID, messageIndex);
+        uint64 uptime = totalUptime - $._posValidatorInfo[validationID].lastClaimMinUptime;
+
+        // If no rewards have yet been claimed, use the validator's start time
+        uint64 lastClaimTime = $._posValidatorInfo[validationID].lastClaimTime;
+        if (lastClaimTime == 0) {
+            lastClaimTime = validator.startedAt;
+        }
+        // Validate the uptime for this claim. Given that all previous claims have been similarly validated,
+        // this is equivalent to validating the uptime of the entire validation period up to this point, due
+        // to the linearity of the uptime threshold calculation.
+        if (!_validateSufficientUptime(uptime, lastClaimTime, claimTime)) {
+            revert ValidatorIneligibleForRewards(validationID);
+        }
+
+        uint256 reward = $._rewardCalculator.calculateReward({
+            stakeAmount: weightToValue(validator.startingWeight),
+            validatorStartTime: lastClaimTime,
+            stakingStartTime: lastClaimTime,
+            stakingEndTime: claimTime,
+            uptimeSeconds: uptime,
+            initialSupply: 0,
+            endSupply: 0
+        });
+
+        $._posValidatorInfo[validationID].lastClaimMinUptime =
+            (claimTime - validator.startedAt) * UPTIME_REWARDS_THRESHOLD_PERCENTAGE / 100;
```

Comment:
I understand that we want to "consume" from the totalUptime only what is needed to claim the reward. This allows the next claim to be more forgiving. However, it is now possible for claimTime - lastClaimTime to be less than totalUptime - lastClaimMinUptime in the next claim (i.e., an uptime that is longer than the claim period). This is not a problem, but seems like it is breaking an invariant at first glance. It may be worth documenting this.